daemon/algod/api/server/v2/handlers.go

```diff
@@ -0,0 +1,420 @@
+			if err == nil {
+				creator = creatorAddr.String()
+			} else {
+				// Asset may have been deleted, so we can no
+				// longer fetch the creator
+				creator = ""
+			}
+
+			holding := generated.AssetHolding{
+				Amount:   holding.Amount,
+				AssetId:  uint64(curid),
+				Creator:  creator,
+				IsFrozen: holding.Frozen,
+			}
+
+			assets = append(assets, holding)
+		}
+	}
+
+	createdAssets := make([]generated.Asset, 0)
+	if len(record.AssetParams) > 0 {
+		for idx, params := range record.AssetParams {
+			assetParams := generated.AssetParams{
+				Creator:       address,
+				Total:         params.Total,
+				Decimals:      uint64(params.Decimals),
+				DefaultFrozen: &params.DefaultFrozen,
+				MetadataHash:  byteOrNil(params.MetadataHash[:]),
+				Name:          strOrNil(params.AssetName),
+				UnitName:      strOrNil(params.UnitName),
+				Url:           strOrNil(params.URL),
+				Clawback:      addrOrNil(params.Clawback),
+				Freeze:        addrOrNil(params.Freeze),
+				Manager:       addrOrNil(params.Manager),
+				Reserve:       addrOrNil(params.Reserve),
+			}
+			asset := generated.Asset{
+				Index:  uint64(idx),
+				Params: assetParams,
+			}
+			createdAssets = append(createdAssets, asset)
+		}
+	}
+
+	var apiParticipation *generated.AccountParticipation
+	if record.VoteID != (crypto.OneTimeSignatureVerifier{}) {
+		apiParticipation = &generated.AccountParticipation{
+			VoteParticipationKey:      byteOrNil(record.VoteID[:]),
+			SelectionParticipationKey: byteOrNil(record.SelectionID[:]),
+			VoteFirstValid:            numOrNil(uint64(record.VoteFirstValid)),
+			VoteLastValid:             numOrNil(uint64(record.VoteLastValid)),
+			VoteKeyDilution:           numOrNil(uint64(record.VoteKeyDilution)),
+		}
+	}
+
+	response := generated.AccountResponse{
+		Type:                        nil,
+		Round:                       uint64(lastRound),
+		Address:                     addr.String(),
+		Amount:                      amount.Raw,
+		PendingRewards:              pendingRewards.Raw,
+		AmountWithoutPendingRewards: amountWithoutPendingRewards.Raw,
+		Rewards:                     record.RewardedMicroAlgos.Raw,
+		Status:                      record.Status.String(),
+		RewardBase:                  &record.RewardsBase,
+		Participation:               apiParticipation,
+		CreatedAssets:               &createdAssets,
+		Assets:                      &assets,
+	}
+
+	return ctx.JSON(http.StatusOK, response)
+}
+
+// GetBlock gets the block for the given round.
+// (GET /v2/blocks/{round})
+func (v2 *Handlers) GetBlock(ctx echo.Context, round uint64, params generated.GetBlockParams) error {
+	handle, err := getCodecHandle(params.Format)
+	if err != nil {
+		return returnError(ctx, http.StatusBadRequest, err, errFailedParsingFormatOption, v2.Log)
+	}
+
+	// TODO: What is raw block bytes, should I use that instead?
+	//blockbytes, err := rpcs.RawBlockBytes(v2.Node.Ledger(), basics.Round(round))
```

Comment:
v1 has a mode `/block/{}?raw=1` that returns the msgpack blob for a block, the same bytes (using the same server side code, also "Content-Type": rpcs.LedgerResponseContentType) as the catchup interface. Indexer and Algobot need this. Without `?raw=1` it returns JSON per the rest of the API spec.

Reply:
This feature stays. The question here is about the certs. Do you use them?

Reply:
I don't use the certs, but we should keep them in by default to be true to the intent of "here's the raw data (all of it)". This endpoint was added because the original API block fetch was missing data. I don't think it's worth adding the complexity of optionally stripping them out because I expect most users of this endpoint to be talking to a local `algod` either within one server or at least within the same datacenter. No bandwidth problem. If you want to get this data remotely, then go to a public archival relay node and GET from the catchup endpoint.

Reply:
I was talking about optionally adding it in but do you expect the msgpack and json answers to return the same thing for this call? 

Also, how do you expect people to contact the catchup endpoint? it's not exactly publicized

Reply:
I'm sure the msgpack case should return everything. The only people who will bother with that mode want everything or can deal with everything. The JSON API can be more tailored to other app developers.

Reply:
Based on this conversation I'll include the cert when the user requests `format=msgp`, with the following change:
```
	response := struct {
		Block bookkeeping.Block     `json:"omitempty"`
		Cert  agreement.Certificate `json:"omitempty"`
	} {
		Block: block,
	}

	// If we're using the messagepack encoding, also include the certificates.
	if handle == protocol.CodecHandle {
		response.Certs = cert
	}
```

I'm not using `RawBlockBytes`

Reply:
@rotemh , we need to keep the certs as an option, especially for the raw format.
When we would need to automate the creation of blocks servers, it would be needed.

Reply:
@tsachiherman is the option critical or the fact that it will be there?

Reply:
@rotemh the option is not *currently* being used, but I was expecting to use this feature when implementing the block storage servers. If it's too big of a change for now, it's ok to loose that ability, and we could work down the road on bringing it back.

Reply:
@tsachiherman could you give me some specific requirements for what you need?

With the current implementation `format=msgp` should provide the certs. The format is a little weird though:
```
{
    "block": "<json/msgp encoded block object>",
    "cert": "<msgp encoded cert object>"
}
```

So the same data is available but there is an extra step to decode the field.

Reply:
Yes, I can. When we would want to have an external block server, we need a way to extract the blocks from the algod instance in a way where we can serve that to another algod which wouldn't be able to tell the difference.
In practice we want to make sure that whatever we have in the database, is being exported without any decoding/encoding. That "blob" would then be stored in S3 ( or others ) and served as a file.
Adding any msgpack decoding/encoding might modify the content, which is what we'd like to avoid.
( it's not just performance issue - it might also modify the correctness of the process )

Reply:
Can we just keep the current behavior where `/block/{blocknumber}?raw=1` returns the exact same data as the catchup endpoint?

Reply:
The catchup endpoint is definitionally all the data anyone could need. It's already done. It's simple. Let's do that?

Reply:
@algobolson correct me if I'm wrong but the new endpoint is just an extension of the old one. The endpoint still delivers the block in msgpack format just with the certs as well. Is there any information that indexer is dependent on that is not being returned now?

Reply:
that's not the current implementation and this sounded like a long thread about maybe doing something else so I wanted to reiterate that we should indeed make available exactly the raw data that the current block endpoint does

Reply:
updates have been made based on this discussion